Add op_vault-for-bitcoin-covenants-panel review to bitcoin-review-podcast (#419)
bitcointranscripts/bitcointranscripts: bitcoin-review-podcast/op_vault-for-bitcoin-covenants-panel.md

So really interested to talk about this proposal.

NVK: 00:01:19

Very cool.
We have Ben de Carman, a return guest.

Ben: 00:01:26

Hey, thanks for having me on again.
I'm guessing I'm here because I posted to the mail as well how you could do the CTV optimization with, for DLCs, you can optimize them with Covenants and I showed how you could do it with `OP_VAULT`s as well.

NVK: 00:01:42

Great.
Antoine?

Antoine: 00:01:44

Hey everyone, I'm Antoine Poinsot.
I've been working on building an actual vault architecture, which does not use Covenant for the past two years and a half.
Lately I've been working on other projects, but I guess that's why I'm here.

NVK: 00:02:00

And the man of the hour, James O'Beirne.

James: 00:02:03

Hey guys.
I'm a Bitcoin developer.
I spend a lot of my days working on core, but for the last few years, I've been really interested in custody systems and the promise of vaults.
And last year, I actually implemented a similar vault design on top of `OP_CTV`, which I'm sure we've covered at some point on the show.
But I was kind of dissatisfied with those implementations for various reasons, and so decided to come up with this proposal.

> So just want to say that to make this a lively discussion, you should do your best to rip into this and you won't offend me, kind of regardless of whatever you say,  NVK said, any new proposal for Bitcoin has to be kind of shredded apart and scrutinized.

Reviewer: missing like
```suggestion
So just want to say that to make this a lively discussion, you should do your best to rip into this and you won't offend me, kind of regardless of whatever you say, like NVK said, any new proposal for Bitcoin has to be kind of shredded apart and scrutinized.
```

So just curious,  how do you sort of  see that?

James: 00:18:33

So it's tricky to think about
Because, I think people rightfully raise the point that if in your fee management strategy you need to rely on actually having other coins available that aren't vaulted.
That's a reasonable objection to me, but at the same time, I think there are definitely plausible schemes where you've still got value locked up in that vault.
And so it's very easy to do this kind of atomic swap type thing where someone offers to fund your unvault on the basis of you presenting them with an unvault that pays them out some amount.

NVK: 00:19:09

I can see that as a service.
And I think that's not an objection, per se.
I think this is more just trying to work out the kinks through.
Because it's sort of  a, we're going to have to rethink.
And I'm speaking as somebody who does the harder wallets.
I'm going to have to rethink of , how do I advise the users on how they do their setups and how do they do key management?
Because that's all going to change.
And this is actually fantastic.
Because multi-sig is a shit coin.
okay.
we have a way now to handle the fees.

> Maybe it's, there is an `OP_vAULT` for the `OP_vAULT`.

Reviewer: You have multiple variances of `OP_VAULT` written.

I need to take my money with me and I have to go naked.
How do you actually go through that process, based on this complex stuff and that kind of leads me to the further picking and risks that I'm trying to sort of think through on OpVault are the practical ones.

## OP_VAULT practical applications

NVK: 00:38:42

I think we can still go through a little bit more of the technical stuff in terms the actual, interesting parts of how this works, but the practical stuff, needs a lot of thoughts still.
 how do you manage these keys?
How do you manage the nuclear codes?
Do you do an op-vault, on op-vault, on op-vault?
So  you have some recursive way of  moving stuff around.
How would wallet UXs address the vaults and create all the stuff?
And, what kinds of things you guys have sort of thought through in this sort of practical space?

James: 00:39:30

I think that's a lot of the work that remains to be done.
I mean, I think the framework right now gives the end user a lot of configurability.
And in designing it, I'm trying to avoid any particular configuration that would be just blatantly unsafe and any use that just wouldn't make any sense.

> But at the same time, I mean, things for your recovery path, are you using a static address there or are you varying it over a descriptor, that does have implications.

Reviewer: ```suggestion
But at the same time, I mean, things like  for your recovery path, are you using a static address there or are you varying it over a descriptor?
That does have implications.
```

And then you have your charter, and then you have all your fiduciary nuances of things.
And you're going to have to follow these very traditional ways on how things are custodied.
And then they might de-risk it by using different vendors as well, so splitting the pile.
And then everything needs to be insured.
And often you have insurance of the insurance as well, especially because if you're trying to, and it's in dollars for the next foreseeable near future here at least, this is denominated in dollars, so you have to get reinsured as the price of Bitcoin goes up and down.
And, they're gonna try to understand your vault, your security,  that's why you use custodians that already have a lot of , provable experience on what they've done, Fidelity or Coinbase or whatever.
And it costs a lot.
The insurance on this stuff is obnoxious.
Because this is a very easy to steal asset.
its best feature is its solubility and, fungibility and transportability.
So how do you do that cheaper, which everybody cares about, and safer
And I think this is great for enterprise that has to deal with insurance.
And soon enough, we're going to see even individuals seeking insurance

> That's what the guys from  Anchor Watch and things  that are trying to do.

Reviewer: example
```suggestion
That's what the guys from AnchorWatch and things like that are trying to do.
```

We're just gonna calculate the Merkle tree, which would still be a little complicationally expensive, but not nearly as enough.
And it makes it so much simpler in the protocol as well, where we need all these round trips back and forth and all this P2P stuff is basically generating address, verify the other card and probably do the same thing and sign a transaction.

NVK: 00:51:24

A critic would say, but why not do this on AWS?
And, just,do the payouts in Bitcoin.
why do we need to do this on Bitcoin?
Why do we need this extra stuff?

James: 00:51:39

Trusted third parties or security holes.

Ben: 00:51:41

Yeah, exactly.
you could deposit in the Coinbase and do the bet and then, withdraw, but we saw FTX worked out.
We saw, the other thousand before them worked out as well.
we need to kind of wait to do this natively.
And I think there is demand  and the shitcoin space,  a lot of people are using , all their trading stuff on there That's  a uni swap and blah blah blah just to  trade stuff quote-unquote trustlessly 

> So there actually is demand for this and it seems I'd rather people use Bitcoin than shitcoins to do it So I hope...

Reviewer: This is an example of self-correction
```suggestion
So there actually is demand for this and I'd rather people use Bitcoin than shitcoins to do it.
```

It's gonna be activation.
Is the idea is , the problem is, I think Speedy Trial  pissed off enough people.
Yep.
And it was kind of  felt rammed through by the cabal of core developers.
So there is that bad taste in the mouth of a lot of economic nodes out there.
The dick butts certainly did not help.
And I still think we don't have a good way of activating stuff.
And we sort of , we're a little PTSD from things.
So it's not going to be upvote.
It's going to be , I don't want to change Bitcoin.
Right.
So and I think a lot of the people who criticize changing Bitcoin, they are correcting the sentiment of not wanting to change Bitcoin.
But they don't also don't understand that  software doesn't live forever without gardening.
the stuff that's running on your computer today doesn't run tomorrow.
And you need the people who understand the code base to sort of upgrade it for new hardware, for new concerns, for, Unix time problems.
And there's all these things that need to change.
when my good friend Steve says, fire the devs, I understand why it feels that way.
And I kind of agree in a way because,  core, I addressed this in the last episode.

> It kind of feels  this, it's  Internet Explorer has 99% of the market.

Reviewer: You see how removal of "like" sometimes distorts the sentence?
